File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <algorithm>
#include <map>
#include <vector>

using namespace std;

int solution(int k, vector<int> tangerine) {
  int answer = 0;
  vector<int> cnts;
  map<int, int> cnt_map;

  for (int& size : tangerine) ++cnt_map[size];
  for (auto& [size, cnt] : cnt_map) cnts.push_back(cnt);
  sort(cnts.begin(), cnts.end(), greater());
  for (int cnt : cnts) {
    k -= cnt;
    ++answer;
    if (k <= 0) break;
  }
  return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

class Solution {

    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> countMap = new HashMap<>();

        for (int size : tangerine) {
            countMap.put(size, countMap.getOrDefault(size, 0) + 1);
        }

        ArrayList<Integer> counts = new ArrayList<>(countMap.values());
        counts.sort(Collections.reverseOrder());

        for (int count : counts) {
            k -= count;
            answer++;
            if (k <= 0) {
                break;
            }
        }

        return answer;
    }
}